types/mithril/index.d.ts

```diff
@@ -42,18 +42,18 @@ declare namespace Mithril {
     }
 
     interface Hyperscript {
+        /** Creates a virtual element (Vnode). */
+        (selector: string, ...children: Children[]): Vnode<any, any>;
```

Comment:
This seems like it increases the chance of picking a non-generic overload over a generic overload. I'm not sure the current tests are specific enough to verify this doesn't cause unwanted behavior. Can you add a test that uses `// $ExpectType` to verify generic overloads are still selected correctly?

Reply:
In my case it was always picking the last overload of `(selector: string, ...children: Children[]): Vnode<any, any>;` and not giving me any specific errors until I reversed the order. if any attributes would be typed wrong it would pick this overload and say that m expected a string without telling me which attribute was mistyped. I don't know anything about testing typings so I doubt I'll get around to creating tests for this. I just copy pasted the file into my project and made these changes so it works for me.